core/src/main/java/tech/pegasys/eth2signer/core/http/handlers/SignForIdentifierHandler.java

```diff
@@ -51,11 +51,11 @@ public SignForPublicKeyHandler(
   @Override
   public void handle(RoutingContext routingContext) {
     final RequestParameters params = routingContext.get("parsedParameters");
-    final String publicKey = params.pathParameter("publicKey").toString();
-    final Optional<ArtifactSigner> signer = signerProvider.getSigner(publicKey);
+    final String identifier = params.pathParameter("identifier").toString();
+    final Optional<ArtifactSigner> signer = signerProvider.getSigner(identifier);
     if (signer.isEmpty()) {
-      LOG.error("Unable to find an appropriate signer for request: {}", publicKey);
-      routingContext.fail(404);
+      LOG.trace("Unable to find an appropriate signer for request: {}", identifier);
+      routingContext.next();
```

Comment:
why doing next() instead of failing with 404?

Reply:
So that we can fall through the other instance of this handler. We have one registered with a ArtifactSignerProvider that handles BLS keys and one that has a secp ArtifactSignerProvider. This way if the key isn't a BLS key we will try find it as a secp key. And if that fails vertx will just return a 404 anyway.